### PR TITLE
Updated for Swift 4

### PR DESCRIPTION
**Pull request by sidmani**

### CODE REVIEW

**Comment by EvsenevDev** [comment]
@sidmani Cool! Thanks.

**Comment by EvsenevDev** [comment]
@sidmani what about Cartfile?

File: Example/Podfile.lock [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-  - RxGesture (1.1.0):
-    - RxCocoa (~> 3.6)
-    - RxSwift (~> 3.6)
-  - RxSwift (3.6.1)
+  - RxCocoa (4.0.0-alpha.1):
+    - RxSwift (~> 4.0.0-alpha.1)
+  - RxGesture (1.1.1):
+    - RxCocoa (~> 4.0.0-alpha.0)
+    - RxSwift (~> 4.0.0-alpha.0)
+  - RxSwift (4.0.0-alpha.1)
```
Comment (jegnux) [inline Example/Podfile.lock]: There's a `4.0.0-beta.0` of RxSwift now. Can you update for it please?

File: Pod/Classes/iOS/UILongPressGestureRecognizer+RxGesture.swift [hunk 2]
```diff
@@ -22,17 +22,20 @@ import UIKit
...
+    public static var minimumPressDuration: CFTimeInterval = 0.5
+    public static var allowableMovement: CGFloat = 10
+    public static var configuration: ((UILongPressGestureRecognizer, RxGestureRecognizerDelegate) -> Void)? = nil
 }
 
+fileprivate typealias Defaults = LongPressGestureRecognizerDefaults
+
 /// A `GestureRecognizerFactory` for `UITapGestureRecognizer`
 public struct LongPressGestureRecognizerFactory: GestureRecognizerFactory {
+    /// Default values for `UILongPressGestureRecognizer` configuration
```
Comment (jegnux) [inline Pod/Classes/iOS/UILongPressGestureRecognizer+RxGesture.swift]: This comment looks misplaced.

File: RxGesture.podspec [hunk 3]
```diff
@@ -29,8 +29,8 @@ Pod::Spec.new do |s|
   s.ios.source_files      = 'Pod/Classes/iOS/*.swift'
   s.osx.source_files      = 'Pod/Classes/OSX/*.swift'
   
-  s.dependency 'RxSwift', '~> 3.6'
-  s.dependency 'RxCocoa', '~> 3.6'
+  s.dependency 'RxSwift', '~> 4.0.0-alpha.0'
+  s.dependency 'RxCocoa', '~> 4.0.0-alpha.0'
```
Comment (jegnux) [inline RxGesture.podspec]: I'm not a pod expert but wouldn't it work with `'RxSwift', '~> 4.0'`? I'm not sure it includes alpha/beta releases.

**Review by jegnux — CHANGES_REQUESTED** [review]
Thanks for your PR. Though, I have minor changes I would like to see. Also, as mentioned above, can you update cartfile to use latest RxSwift version please?

File: RxGesture.podspec (hunk 3 above)
Comment (sidmani) [reply RxGesture.podspec]: CocoaPods doesn't resolve "4.0.0-beta.0" as satisfying "\~> 4.0" so I think "= 4.0.0-beta.0" is the best we can do for now.

**Comment by sidmani** [comment]
Made the changes.

File: Example/Podfile.lock (hunk 1 above)
Comment (jegnux) [reply Example/Podfile.lock]: @sidmani Can you `pod update` please in order to update the `Podfile.lock` file?

**Review by jegnux — APPROVED** [review]

**Pull request merged**